aggregator/src/aggregator/error.rs

```diff
@@ -232,3 +234,63 @@ impl Display for BatchMismatch {
         )
     }
 }
+
+/// Inspect the provided `ping_pong_error`, log it, increment the [`Counter`] with appropriate
+/// labels, and return a suitable [`PrepareError`].
+pub(crate) fn handle_ping_pong_error(
+    task_id: &TaskId,
+    peer_role: Role,
+    report_id: &ReportId,
+    ping_pong_error: PingPongError,
+    aggregate_step_failure_counter: &Counter<u64>,
+) -> PrepareError {
+    let (error_desc, value, prepare_error) = match ping_pong_error {
+        PingPongError::VdafPrepareInit(_) => (
+            "Couldn't helper_initialize report share".to_string(),
+            "prepare_init_failure".to_string(),
+            PrepareError::VdafPrepError,
+        ),
+        PingPongError::VdafPreparePreprocess(_) => (
+            "Couldn't compute prepare message".to_string(),
+            "prepare_message_failure".to_string(),
+            PrepareError::VdafPrepError,
+        ),
+        PingPongError::VdafPrepareStep(_) => (
+            "Prepare step failed".to_string(),
+            "prepare_step_failure".to_string(),
+            PrepareError::VdafPrepError,
+        ),
+        PingPongError::CodecPrepShare(_) => (
+            format!("Couldn't decode {peer_role} prepare share"),
+            format!("{peer_role}_prep_share_decode_failure"),
+            PrepareError::UnrecognizedMessage,
+        ),
+        PingPongError::CodecPrepMessage(_) => (
+            format!("Couldn't decode {peer_role} prepare message"),
+            format!("{peer_role}_prep_message_decode_failure"),
+            PrepareError::UnrecognizedMessage,
+        ),
+        ref error @ PingPongError::StateMismatch(_, _) => (
+            format!("{error}"),
+            format!("{peer_role}_ping_pong_message_state_mismatch"),
+            // TODO(timg): is this the right error if state mismatch?
```

Comment:
This needs to be resolved: `VdafPrepError` is correct in this case, because the DAP text says that any time a VDAF-level ping-pong routine yields state `Rejected()`, the aggregator should construct a `PrepareResp` with `VdafPrepError`. So for that reason, the cases above where we return `UnrecognizedMessage` need to be fixed. It's a shame, because in libprio we actually have specific knowledge of what went wrong, but per the specification we have to discard that information and flatten everything into `VdafPrepError`. At least we can still increment a specific Prometheus metric.